projects/packages/assets/src/class-assets.php

```diff
@@ -92,7 +92,7 @@ public function script_add_async( $tag, $handle ) {
 		}
 
 		if ( in_array( $handle, $this->defer_script_handles, true ) ) {
-			return preg_replace( '/^<script /i', '<script defer ', $tag );
+			return preg_replace( '/<script(.*) src=/i', '<script defer$1 src=', $tag );
```

Comment:
This regex would fail to work properly on input like `<scriptxyz src=` or `<script>no newline in here</script><script src=`.

This should fix it
```suggestion
			return preg_replace( '/<script( [^>]*)? src=/i', '<script defer$1 src=', $tag );
```

Reply:
Good call! I've added a test for that behavior in https://github.com/Automattic/jetpack/pull/23928/commits/6aa03816feeafe5be4dd87acb51c5f5543974e98.